urwid/widget/widget.py

```diff
@@ -576,6 +570,69 @@ def _repr_words(self):
     def _repr_attrs(self):
         return {}
 
+    def keypress(self, size: tuple[()] | tuple[int] | tuple[int, int], key: str) -> str | None:
+        """Keyboard input handler.
+
+        :param size: See :meth:`Widget.render` for details
+        :type size: tuple[()] | tuple[int] | tuple[int, int]
+        :param key: a single keystroke value; see :ref:`keyboard-input`
+        :type key: str
+        :return: ``None`` if *key* was handled by *key* (the same value passed) if *key* was not handled
+        :rtype: str | None
+        """
+        if not self.selectable():
+            if hasattr(self, "logger"):
+                self.logger.debug(f"Widget {self!r} is not selectable")
```

Comment:
Is this a clearer message?
```suggestion
                self.logger.debug(f"keypress sent to non selectable widget {self!r}")
```